src/client/dfs/duns.c

```diff
@@ -879,14 +927,61 @@ duns_create_path(daos_handle_t poh, const char *path, struct duns_attr_t *attrp)
 		}
 #endif
 
+		D_STRNDUP(dir, path, path_len);
+		if (dir == NULL)
+			return ENOMEM;
+
+		dirp = dirname(dir);
+
 		/** create a new directory if POSIX/MPI-IO container */
 		rc = mkdir(path, mode);
 		if (rc == -1) {
 			rc = errno;
 
-			D_ERROR("Failed to create dir %s: %s\n", path, strerror(rc));
+			D_ERROR("Failed to create dir %s: %d (%s)\n", path, rc, strerror(rc));
 			return rc;
 		}
+
+		rc = statfs(dirp, &fs);
+		if (rc == -1) {
+			int err = errno;
+
+			D_ERROR("Failed to statfs dir %s: %d (%s)\n", dirp, err, strerror(err));
+			D_FREE(dir);
+			return err;
+		}
+
+		/* Open the parent directory for the new container so we can call a dfuse iotcl
+		 * to discover the user running dfuse.
+		 */
+		if (fs.f_type == FUSE_SUPER_MAGIC) {
+			int fd;
+
+			fd = open(dirp, O_RDONLY | O_DIRECTORY | O_NOFOLLOW);
+			if (fd == -1) {
+				int err = errno;
+
+				D_ERROR("Dfuse open failed '%s': %d (%s)\n", dirp, err,
+					strerror(err));
+				D_FREE(dir);
+				return err;
+			}
+
+			rc = ioctl(fd, DFUSE_IOCTL_DFUSE_USER, &dur);
```

Comment:
is it possible to change the ioctl to only return the uid instead of struct dfuse_user_reply (both uid and gid)? or is it used somewhere else where gid is needed?
it is called *_USER after all?

Reply:
actually doesn't matter.  still won't be able to remove the dependency on the header for the IOCTL value itself.. it's fine i guess.